source/common/http/conn_manager_utility.cc

```diff
@@ -289,11 +289,18 @@ void ConnectionManagerUtility::cleanInternalHeaders(
     RequestHeaderMap& request_headers, bool edge_request,
     const std::list<Http::LowerCaseString>& internal_only_headers) {
   if (edge_request) {
+    // Headers to be stripped from edge requests, i.e. to sanitize so
+    // clients can't inject values.
     request_headers.removeEnvoyDecoratorOperation();
     request_headers.removeEnvoyDownstreamServiceCluster();
     request_headers.removeEnvoyDownstreamServiceNode();
+    if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.sanitize_original_path")) {
+      request_headers.removeEnvoyOriginalPath();
+    }
   }
 
+  // Headers to be stripped from edge *and* upstream server requests
+  // - these headers should only ever be seen or set by envoy itself.
```

Comment:
This is not really accurate. Essentially this block is removing all headers where there is some indication that it is external. This could be a true external connection, or XFF indicates that it's external origin.

I'm still pretty confused about your internal setup that is causing this to be a problem. If you are more comfortable let's talk in Slack offline.